src/analyze_model.py

```diff
@@ -0,0 +1,195 @@
+        output (num_r_out, num_c_out), build a map that can convert
+        inputs row/cols to bound box rows/cols
+    Args:
+        num_r_in (int): number of rows in input
+        num_c_in (int): number of columns in input
+        num_r_out (int): number of rows in bounding box output
+        num_c_out (int): number of columns in bounding box output
+    Returns:
+        region_map_r (np.array shape=[num_r_in,]): maps initial input rows
+            to the binned rows of the output r_out = region_map_r[r_in]
+        region_map_c (np.array shape=[num_c_in,]): maps initial input columns
+            to the binned rows of the output c_out = region_map_c[c_in]
+    Updates:
+        N/A
+    Write to file:
+        N/A
+    '''
+    # divide up input into regions
+    delta_r = np.floor(float(num_r_in) / float(num_r_out))
+    delta_c = np.floor(float(num_c_in) / float(num_c_out))
+    # build a map from input rows to output rows
+    region_map_r = np.repeat(np.arange(num_r_out), delta_r)[0:num_r_in]
+    region_map_c = np.repeat(np.arange(num_c_out), delta_c)[0:num_c_in]
+    return region_map_r, region_map_c
+
+
+def output_vec_2_to_bb_output(y, reg_coors, num_r, num_c):
+    '''
+    Description:
+        Given an output vector for each label, put it into
+        the output matrix. Each label should go into the
+        correct region.
+    Args:
+        y (np.array, size=[num_labels, n]): all of the labled output
+            vectors, where n depends on the number of classes/anchor boxes
+        num_r (int): number of rows of output bounding box matrix
+        num_c (int): number of cols of output bounding box matrix
+        reg_coors (np.array(num_trees, 2): the coordinates of the localized
+            objects
+    Returns:
+        output (np.array, size=[num_r_out, num_c_out, n]): output array
+            for each bounding box regional. n is the length of the output
+            array, n = (5+num_classes). Note, no anchor boxes!!!
+    Updates:
+        N/A
+    Write to file:
+        N/A
+    '''
+    num_2_fill = len(reg_coors)
+    output = np.zeros((num_r, num_c, np.shape(y)[1]))
+    for index in np.arange(num_2_fill):
+        output[reg_coors[index, 0], reg_coors[index, 1], :] = y[index, :]
+    return output
+
+
+def parse_labels(filename):
+    '''
+    Description:
+        Parse the lableImg files (assumes YOLO format)
+    Args:
+        filename (str): path to label
+    Returns:
+        label_mat (np.array, size=[num_labels,  5]): output matrix
+            for each label. Each row is separate label
+            with values ('class', x, y, w, h)
+    Updates:
+        N/A
+    Write to file:
+        N/A
+    '''
+    labels = []
+    # open file and grab lines
+    labfile = open(filename, "r")
+    for line in labfile.readlines():
+        data = line.split()
+        labels.append(data)
+    # convert to a more useful format
+    num_labels = len(labels)
```

Comment:
Is this guaranteed based on the way the inputs are fed in? Is it possible you have something that looks like `[[1], [2,3]]`? If so, how will you catch that?